src/core/weights.js

```diff
@@ -165,3 +166,72 @@ export function toJSON(weights: Weights): WeightsJSON {
 export function fromJSON(json: WeightsJSON): Weights {
   return parser.parseOrThrow(json);
 }
+
+export type WeightsComparison = {|
+  +weightsAreEqual: boolean,
+  +uniqueNodesInFirst: $ReadOnlyArray<[NodeAddressT, NodeWeight]>,
```

Comment:
The name `uniqueNodesInFirst` is confusing, since we don't have any `Node`s in this data structure at all

How about the simpler form like this:

`export type NodeWeightDiff = {|+first: NodeWeight | null, +second: NodeWeight | null|}`
then insetad of having uniqueNodesInFirst, uniqueNodesInSecond, nodeTuplesWithDifferences, there can be a single field of diffs

Reply:
accepted